crypto/rand/drbg_lib.c

```diff
@@ -1090,10 +1091,13 @@ static int drbg_add(const void *buf, int num, double randomness)
          * inevitably. So we use a trick to mix the buffer contents into
          * the DRBG state without forcing a reseeding: we generate a
          * dummy random byte, using the buffer content as additional data.
+         * Note: This won't work with RAND_DRBG_FLAG_CTR_NO_DF.
```

Comment:
Aehm, sorry, I wrote this comment when I saw this in drbg_ctr_generate:
```
    if (adin != NULL && adinlen != 0) {
        if (!ctr_update(drbg, adin, adinlen, NULL, 0, NULL, 0))
            return 0;
        /* This means we reuse derived value */
        if ((drbg->flags & RAND_DRBG_FLAG_CTR_NO_DF) == 0) {
            adin = NULL;
            adinlen = 1;
        }
```
I only saw adin = NULL, but I overlooked that adin/adinlen gets already used in the ctr_update above.
So I would like to drop this comment before merging, if you don't mind.

Reply:
no, forget it. the problem with CTR_NO_DF is that ctr_update uses only 16 bytes of it.
So the comment is right, although I wrote it for the wrong reason....